In [3]:
import pandas as pd

In [5]:
import os

print(os.getcwd())

C:\Projects\saudi-job-market-analysis


In [6]:
df = pd.read_csv("data/raw/postings.csv")

In [7]:
df.head()

,job_id,company_name,title,description,max_salary,pay_period,location,company_id,views,med_salary,...,skills_desc,listed_time,posting_domain,sponsored,work_type,currency,compensation_type,normalized_salary,zip_code,fips
0,921716,Corcoran Sawyer Smith,Marketing Coordinator,Job descriptionA leading real estate firm in N...,20.0,HOURLY,"Princeton, NJ",2774458.0,20.0,NaN,...,Requirements: \n\nWe are seeking a College or ...,1.713398e+12,NaN,0,FULL_TIME,USD,BASE_SALARY,38480.0,8540.0,34021.0
1,1829192,NaN,Mental Health Therapist/Counselor,"At Aspen Therapy and Wellness , we are committ...",50.0,HOURLY,"Fort Collins, CO",NaN,1.0,NaN,...,NaN,1.712858e+12,NaN,0,FULL_TIME,USD,BASE_SALARY,83200.0,80521.0,8069.0
2,10998357,The National Exemplar,Assitant Restaurant Manager,The National Exemplar is accepting application...,65000.0,YEARLY,"Cincinnati, OH",64896719.0,8.0,NaN,...,We are currently accepting resumes for FOH - A...,1.713278e+12,NaN,0,FULL_TIME,USD,BASE_SALARY,55000.0,45202.0,39061.0
3,23221523,"Abrams Fensterman, LLP",Senior Elder Law / Trusts and Estates Associat...,Senior Associate Attorney - Elder Law / Trusts...,175000.0,YEARLY,"New Hyde Park, NY",766262.0,16.0,NaN,...,This position requires a baseline understandin...,1.712896e+12,NaN,0,FULL_TIME,USD,BASE_SALARY,157500.0,11040.0,36059.0
4,35982263,NaN,Service Technician,Looking for HVAC service tech with experience ...,80000.0,YEARLY,"Burlington, IA",NaN,3.0,NaN,...,NaN,1.713452e+12,NaN,0,FULL_TIME,USD,BASE_SALARY,70000.0,52601.0,19057.0


In [8]:
df.shape

(123849, 31)

In [9]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 123849 entries, 0 to 123848
Data columns (total 31 columns):
 #   Column                      Non-Null Count   Dtype  
---  ------                      --------------   -----  
 0   job_id                      123849 non-null  int64  
 1   company_name                122130 non-null  str    
 2   title                       123849 non-null  str    
 3   description                 123842 non-null  str    
 4   max_salary                  29793 non-null   float64
 5   pay_period                  36073 non-null   str    
 6   location                    123849 non-null  str    
 7   company_id                  122132 non-null  float64
 8   views                       122160 non-null  float64
 9   med_salary                  6280 non-null    float64
 10  min_salary                  29793 non-null   float64
 11  formatted_work_type         123849 non-null  str    
 12  applies                     23320 non-null   float64
 13  original_listed_time     

In [10]:
df.isnull().sum()

job_id                             0
company_name                    1719
title                              0
description                        7
max_salary                     94056
pay_period                     87776
location                           0
company_id                      1717
views                           1689
med_salary                    117569
min_salary                     94056
formatted_work_type                0
applies                       100529
original_listed_time               0
remote_allowed                108603
job_posting_url                    0
application_url                36665
application_type                   0
expiry                             0
closed_time                   122776
formatted_experience_level     29409
skills_desc                   121410
listed_time                        0
posting_domain                 39968
sponsored                          0
work_type                          0
currency                       87776
c

In [11]:
df.columns

Index(['job_id', 'company_name', 'title', 'description', 'max_salary',
       'pay_period', 'location', 'company_id', 'views', 'med_salary',
       'min_salary', 'formatted_work_type', 'applies', 'original_listed_time',
       'remote_allowed', 'job_posting_url', 'application_url',
       'application_type', 'expiry', 'closed_time',
       'formatted_experience_level', 'skills_desc', 'listed_time',
       'posting_domain', 'sponsored', 'work_type', 'currency',
       'compensation_type', 'normalized_salary', 'zip_code', 'fips'],
      dtype='str')

In [12]:
df["location"].value_counts().head(20)

location
United States                      8125
New York, NY                       2756
Chicago, IL                        1834
Houston, TX                        1762
Dallas, TX                         1383
Atlanta, GA                        1363
Boston, MA                         1176
Austin, TX                         1083
Charlotte, NC                      1075
Phoenix, AZ                        1059
Washington, DC                      985
Los Angeles, CA                     972
San Francisco, CA                   884
New York City Metropolitan Area     837
Seattle, WA                         818
San Diego, CA                       790
Denver, CO                          787
Philadelphia, PA                    711
Tampa, FL                           659
Miami, FL                           643
Name: count, dtype: int64

In [13]:
df[df["location"].str.contains("Saudi", case=False, na=False)]

,job_id,company_name,title,description,max_salary,pay_period,location,company_id,views,med_salary,...,skills_desc,listed_time,posting_domain,sponsored,work_type,currency,compensation_type,normalized_salary,zip_code,fips


In [14]:
df[df["location"].str.contains("Riyadh", case=False, na=False)]

,job_id,company_name,title,description,max_salary,pay_period,location,company_id,views,med_salary,...,skills_desc,listed_time,posting_domain,sponsored,work_type,currency,compensation_type,normalized_salary,zip_code,fips


In [15]:
df[df["location"].str.contains("Jeddah", case=False, na=False)]

,job_id,company_name,title,description,max_salary,pay_period,location,company_id,views,med_salary,...,skills_desc,listed_time,posting_domain,sponsored,work_type,currency,compensation_type,normalized_salary,zip_code,fips


In [17]:
import sys

print(sys.executable)

C:\Projects\saudi-job-market-analysis\.venv\Scripts\python.exe


In [18]:
df = pd.read_csv("data/raw/postings.csv")

In [19]:
df = pd.read_csv("data/raw/Jadarat_data.csv")


In [20]:
df.head()

,job_title,job_date,job_desc,job_tasks,comp_name,comp_no,comp_type,comp_size,eco_activity,qualif,region,city,benefits,contract,positions,job_post_id,exper,gender
0,محاسب,27/05/1444,['إعداد وتنظيم مستندات الصرف ومتابعة تحصيل الإ...,[' تدقيق المطالبات المالية والتأكد من اكتمال...,شركة مقر العالم للسفريات,1-317262,خاص,متوسطة فئة أ,أنشطة وكالات السياحة والسفر,"['Language data', 'اللغة الانجليزية', 'محترف']",الرياض,AR RIYADH...,"['Salary', '5000.0']",دوام كامل,0 / 1,20202026350419,0 Years,both
1,بائع,27/05/1444,['بيع مجموعة من السلع والخدمات للعملاء، وتوفير...,"[' بيع مجموعة من السلع والخدمات للعملاء.', '...",شركة عالم الكهرباء للمقاولات,4-1324428,خاص,متوسطة فئة ب,تركيب انظمة التبريد وتكييف الهواء وصيانتها واص...,NaN,المنطقة الشرقية,AD DAMMAM...,"['Salary', '5000.0']",دوام كامل,0 / 3,20202026350389,0 Years,both
2,أخصائي عمليات موارد بشرية,27/05/1444,['تنفيذ الإجراءات والأنظمة والنماذج الخاصة بمر...,[' تنفيذ الإجراءات والأنظمة والنماذج الخاصة ...,شركه دار السلام,1-155294,خاص,متوسطة فئة أ,ترميمات المباني السكنية والغير سكنية,"['Language data', 'الانجليزيه', 'محترف']",الرياض,ATH THUMA...,"['Salary', '4000.0']",دوام كامل,0 / 2,20202026350347,2 Years,both
3,ميكانيكي سيارات,27/05/1444,['تشخيص أعطال السيارات وإصلاحها وتنفيذ برامج ا...,[' فحص أداء المعدّات الكهربائية والميكانيكية...,مؤسسة لمكو لغيار الزيوت,8-1925495,خاص,صغيرة فئة ب,NaN,"['Skill data', 'صيانة السيارات وتقييم الاعطال'...",المنطقة الشرقية,AL HUFUF...,"['Salary', '5000.0']",دوام كامل,0 / 10,20202026350219,0 Years,M
4,محاسب,27/05/1444,['إعداد وتنظيم مستندات الصرف ومتابعة تحصيل الإ...,[' تدقيق المطالبات المالية والتأكد من اكتمال...,مؤسسة فكرة اليمامة للمقاولات,1-2356639,خاص,كبيرة,الإنشاءات العامة للمباني السكنية,"['Skill data', 'تحمل ضغط العمل', 'محترف', 'Lan...",الرياض,AR RIYADH...,"['Salary', '5000.0']",دوام كامل,0 / 1,20202026350043,0 Years,both


In [21]:
df.shape

(1470, 18)

In [22]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1470 entries, 0 to 1469
Data columns (total 18 columns):
 #   Column        Non-Null Count  Dtype
---  ------        --------------  -----
 0   job_title     1470 non-null   str  
 1   job_date      1470 non-null   str  
 2   job_desc      1470 non-null   str  
 3   job_tasks     1470 non-null   str  
 4   comp_name     1470 non-null   str  
 5   comp_no       1470 non-null   str  
 6   comp_type     1470 non-null   str  
 7   comp_size     1456 non-null   str  
 8   eco_activity  1380 non-null   str  
 9   qualif        780 non-null    str  
 10  region        1470 non-null   str  
 11  city          1470 non-null   str  
 12  benefits      1470 non-null   str  
 13  contract      1470 non-null   str  
 14  positions     1470 non-null   str  
 15  job_post_id   1470 non-null   int64
 16  exper         1470 non-null   str  
 17  gender        1470 non-null   str  
dtypes: int64(1), str(17)
memory usage: 206.8 KB


In [23]:
df["job_title"].value_counts()

job_title
بائع                       98
محاسب                      89
أخصائي تسويق               54
مساعد إداري                45
موظف استقبال               45
                           ..
أخصائي علوم أحياء دقيقة     1
نجار هياكل خشبية            1
مدير مشتريات                1
مشرف مدخلي البيانات         1
مشغل آلة تشطيب رخام         1
Name: count, Length: 324, dtype: int64

### Insight

The most advertised job title is **بائع** with **98 job postings**, followed by **محاسب** with **89 job postings**.

## Top Cities

In [27]:
df["city"].value_counts().head(10)

city
AR RIYADH...    523
JEDDAH...       245
AD DAMMAM...     78
MAKKAH AL...     72
AL KHUBAR...     54
AL MADINA...     51
ABA ASLAB...     51
AL BAHA...       24
HAIL...          17
AL KHARJ...      16
Name: count, dtype: int64

### Insight

Riyadh has the highest number of job postings (523), followed by Jeddah (245). This suggests that job opportunities are highly concentrated in major cities, especially Riyadh.

In [28]:
df["city"].unique()[:20]

<StringArray>
['AR RIYADH...', 'AD DAMMAM...', 'ATH THUMA...',  'AL HUFUF...',
 'AL BATALI...',    'JEDDAH...',      'HAIL...', 'MAKKAH AL...',
   'GHITHAH...', 'AL MADINA...', 'ABA ASLAB...',     'YANBU...',
   'DHAHRAN...',   'AL AQIQ...',   'AL UYUN...', 'WADI AD D...',
   'TANUMAH...',  'BURAYDAH...',    'BISHAH...', 'AL HAWIYA...']
Length: 20, dtype: str

### Data Quality Note

Some city names contain trailing dots (...) in the original dataset. This issue will be handled during the data cleaning stage.

In [31]:
df["region"].value_counts().head(10)

region
الرياض             622
مكة المكرمة        369
المنطقة الشرقية    219
المدينة المنورة     86
الباحة              46
عسير                42
القصيم              32
حائل                18
تبوك                11
نجران                8
Name: count, dtype: int64

### Insight

Riyadh region has the highest number of job postings (622), followed by Makkah (369) and the Eastern Province (219). This indicates that employment opportunities are concentrated in a few major regions of Saudi Arabia.

In [32]:
df["contract"].value_counts()

contract
دوام كامل     1448
عمل عن بعد      22
Name: count, dtype: int64

In [33]:
df["gender"].value_counts()

gender
both    587
M       480
F       403
Name: count, dtype: int64

### Insight

Most job postings are open to both genders (587), while 480 are targeted at males and 403 at females. This suggests that many employers do not restrict job opportunities by gender.